src/complex.jl

```diff
@@ -0,0 +1,275 @@
+"""
+    iscomplex(x::AbstractVariable)
+
+Return whether a given variable was declared as a complex or real-valued variable.
+By default, all variables are real-valued.
+"""
+iscomplex(x::AbstractVariable) = false
+
+"""
+    isrealpart(x::AbstractVariable)
+
+Return whether the given variable is the real part of a complex-valued variable.
+
+See also [`iscomplex`](@ref iscomplex), [`isimagpart`](@ref isimagpart), [`isconj`](@ref isconj).
+"""
+isrealpart(x::AbstractVariable) = false
+
+"""
+    isimagpart(x::AbstractVariable)
+
+Return whether the given variable is the imaginary part of a complex-valued variable.
+
+See also [`iscomplex`](@ref iscomplex), [`isrealpart`](@ref isrealpart), [`isconj`](@ref isconj).
+"""
+isimagpart(x::AbstractVariable) = false
+
+"""
+    isconj(x::AbstractVariable)
+
+Return whether the given variable is obtained by conjugating a user-defined complex-valued variable.
+
+See also [`iscomplex`](@ref iscomplex), [`isrealpart`](@ref isrealpart), [`isimagpart`](@ref isimagpart).
+"""
+isconj(x::AbstractVariable) = false
+
+"""
+    ordvar(x::Union{AbstractVariable, AbstractVector{<:AbstractVariable}})
+
+Given some (complex-valued) variable that was transformed by conjugation, taking its real part, or taking its
+imaginary part, return the original variable as it was defined by the user.
+
+See also [`conj`](@ref conj), [`real`](@ref), [`imag`](@ref).
+"""
+ordvar(x::AbstractVariable) = x
```

Comment:
Where does this name come from?

Reply:
"Ordinary variable". But `origvar` ("original variable"), `declvar` (declared variable) - or unabbreviated forms - also make sense.

Reply:
Let's use the unabbreviated

Reply:
Done